### PR TITLE
Update DS18S20.cpp and .h

### PR DESCRIPTION
**Pull request by Ogyb**
Added callback on end of conversion.
in DS18S20.h File
public section:

	void RegisterEndCallback(void (*)()); //End conversion "interrupt" function register
	void UnRegisterCallback(); //Delete "interrupt" [end conversion]
private

void (*ReadEndCallback)() = 0;

file DS18s20.h in separate pull...

### CODE REVIEW

File: Sming/Libraries/DS18S20/ds18s20.cpp [hunk 1]
```diff
@@ -255,3 +259,12 @@ uint8_t DS18S20::GetSensorsCount()
 
 }
 
+void DS18S20::RegisterEndCallback(void (*Callback)())
```
Comment (slaff) [inline Sming/Libraries/DS18S20/ds18s20.cpp]: @Ogyb Can you replace the callback with Delegates. This way we can have the old C style callback and also method calling from objects.

**Comment by flexiti** [comment]
@Ogyb 
Not so fast please
Do not make changes that are not compatible with are already running programs! (Eg. sample from sming example, user programs etc.)
At most add new functionality only

**Comment by slaff** [comment]
Please, update in one PR the header and the C++ files. A PR should contain a change that compiles and works on the device.

**Comment by flexiti** [comment]
@Ogyb 

> Its compatible with are already running programs! :)
> I no delete any functionality, only add 2 function and 1 varible.

that's  good :) I saw that one line had left, and I did not notice that  is back
This is library so backwards compatibility is important  :)

**Comment by Ogyb** [comment]
2 file in 1 pull.

**Pull request closed without merging**